mypy/server/deps.py

```diff
@@ -254,6 +254,14 @@ def process_type_info(self, info: TypeInfo) -> None:
                                 target=make_trigger(info.fullname() + '.__init__'))
             self.add_dependency(make_trigger(base_info.fullname() + '.__new__'),
                                 target=make_trigger(info.fullname() + '.__new__'))
+            # If the set of abstract attributes change, this may invalidate class
+            # instantiation, or change the generated error message, since Python checks
+            # class abstract status when creating an instance.
+            self.add_dependency(make_trigger(base_info.fullname() + '.(abstract)'),
+                                target=make_trigger(info.fullname() + '.__init__'))
+            # If the base class abstract attributes change, subclass abstract
+            # attributes need to be recalculated.
+            self.add_dependency(make_trigger(base_info.fullname() + '.(abstract)'))
```

Comment:
Could these be more targeted? I think we shouldn't need dependencies from all the base classes abstract attrs to `__init__`, just from the class itself, and only if it is abstract (since abstract changes triggers the whole class)? And likewise I think the dependency from base abstract attrs is only needed if the base is abstract.

Maybe add a TODO?